src/Context/AuthContext/AuthReducer.js

```diff
@@ -0,0 +1,52 @@
+export const initialState = {
+  foundUser: {
+    _id: null,
+    firstName: null,
+    lastName: null,
+    email: null,
+    cart: [],
+    wishlist: [],
+    id: null,
+  },
+  encodedToken: null,
+};
+
+export const authReducer = (
+  state,
+  { type, payload: { foundUser, encodedToken } }
+) => {
+  switch (type) {
+    case "LOGIN_USER":
+      return {
+        ...state,
+        foundUser: {
+          _id: foundUser._id,
+          firstName: foundUser.firstName,
+          lastName: foundUser.lastName,
+          email: foundUser.email,
+          cart: foundUser.cart,
+          wishlist: foundUser.wishlist,
+          id: foundUser.id,
+        },
+        encodedToken,
+      };
+
+    case "LOGOUT_USER":
+      return {
+        ...state,
+        foundUser: {
+          _id: foundUser._id,
+          firstName: foundUser.firstName,
+          lastName: foundUser.lastName,
+          email: foundUser.email,
+          cart: foundUser.cart,
+          wishlist: foundUser.wishlist,
+          id: foundUser.id,
+        },
+        encodedToken,
+      };
```

Comment:
suggestion: you can have a single case here, "HANDLE_USER", since both the cases are similar and you are passing payload anyway.

Reply:
Done :+1: